### PR TITLE
First pass of renaming 'processor' to 'prepper'.

### PR DESCRIPTION
**Pull request by wrijeff**
*Issue #, if available:* #209

*Description of changes:*
* Used IDE refactor to rename Processor interface and package
* Manually fixed some config yaml files to reference `prepper:` instead of `processor:`

There are still references to "processor" in some of the core files and plugins, will issue follow-up PRs to keep the size managable.

Tested these changes using the dev docker-compose test project, sample log:
```
204  [main] INFO  com.amazon.dataprepper.parser.PipelineParser  – Building preppers for the pipeline [entry-pipeline]
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kowshikn** [comment]
Looks good but should test the docker and archives once. I will test it today and merge it.

**Comment by wrijeff** [comment]
> Looks good but should test the docker and archives once. I will test it today and merge it.

Actually on second thought, this change might require version increase since we're changing what is expected to be in the yaml files (replacing `processor:` with `prepper:`. The current docker-compose examples will likely fail from the changes to `trace_analytics.yml` and `trace_analytics_no_ssl.yml`. 🤔

**Comment by kowshikn** [comment]
Yep.

> > Looks good but should test the docker and archives once. I will test it today and merge it.
> 
> Actually on second thought, this change might require version increase since we're changing what is expected to be in the yaml files (replacing `processor:` with `prepper:`). The current docker-compose examples will likely fail from the changes to `trace_analytics.yml` and `trace_analytics_no_ssl.yml`; we'd have to push these changes to Dockerhub 🤔

Yes. Lets make it work with https://github.com/opendistro-for-elasticsearch/data-prepper/tree/master/examples/dev. Have a story to line up with next docker release.

**Comment by wrijeff** [comment]
Ran `docker-compose up -d` in `/examples/trace-analytics-sample-app` and confirmed the app worked as expected:
```
3264 [main] INFO  com.amazon.dataprepper.pipeline.Pipeline  – Pipeline [entry-pipeline] - Submitting request to initiate the pipeline processing
3364 [entry-pipeline-process-worker-1-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker  –  entry-pipeline Worker: No records received from buffer
5456 [raw-pipeline-process-worker-3-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker  –  raw-pipeline Worker: No records received from buffer
11329 [entry-pipeline-process-worker-1-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker  –  entry-pipeline Worker: Processing 4 records from buffer
11431 [service-map-pipeline-process-worker-2-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker  –  service-map-pipeline Worker: Processing 4 records from buffer
```

**Review by kowshikn — APPROVED** [review]

File: docs/readme/trace_overview.md [hunk 1]
```diff
@@ -13,18 +13,18 @@ The OpenTelemetry source accepts trace data from the OpenTelemetry collector. Th
...
-* *otel_trace_raw_processor* -  This processor is responsible for converting the trace data in [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-proto/tree/master/opentelemetry/proto/trace/v1) to elasticsearch friendly (JSON) docs. These elasticsearch friendly docs have certain additional fields like duration which are not part of the original OpenTelemetry specification. These additional fields are to make the instant kibana dashboards user-friendly.
-* *service_map_stateful* -  This processor performs the required preprocessing on the trace data and build metadata to display the service-map kibana dashboards.
+We have two preppers for the Trace Analytics feature,
+* *otel_trace_raw_processor* -  This prepper is responsible for converting the trace data in [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-proto/tree/master/opentelemetry/proto/trace/v1) to elasticsearch friendly (JSON) docs. These elasticsearch friendly docs have certain additional fields like duration which are not part of the original OpenTelemetry specification. These additional fields are to make the instant kibana dashboards user-friendly.
```
Comment (chenqi0805) [inline docs/readme/trace_overview.md:17]: Is `otel_trace_raw_processor` going to be changed later?

File: data-prepper-plugins/otel-trace-raw-processor/README.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # OTel Trace Raw Processor
```
Comment (chenqi0805) [inline data-prepper-plugins/otel-trace-raw-processor/README.md:1]: Is this going to be changed later? Also applies to the plugin folder name

File: docs/readme/overview.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ The buffer component will act as the layer between the *source* and *sink.* The
...
 
 ### Processor
-Processor component of the pipeline, these are intermediary processing units using which users can filter, transform and enrich the records into desired format before publishing to the sink. The processor is an optional component of the pipeline, if not defined the records will be published in the format as defined in the source. You can have more than one processor, and they are executed in the order they are defined in the pipeline spec.
+Processor component of the pipeline, these are intermediary processing units using which users can filter, transform and enrich the records into desired format before publishing to the sink. The prepper is an optional component of the pipeline, if not defined the records will be published in the format as defined in the source. You can have more than one prepper, and they are executed in the order they are defined in the pipeline spec.
```
Comment (chenqi0805) [inline docs/readme/overview.md:25]: There is mixture of "Processor" and "prepper" in this section including the title. Could be a little confusing.

**Review by chenqi0805 — APPROVED** [review]
Just some questions about the reminiscent "processor" pattern. Nothing blocking

**Comment by wrijeff** [comment]
> Just some questions about the reminiscent "processor" pattern. Nothing blocking

Thanks for the comments; I'll post follow-up PRs for the remaining processor occurrences. 

This PR only includes IDE-driven replacements with the intent that it could be quickly discarded and remade if too many conflicts arose (e.g. branch became too stale). Once the bulk of the changes are merged, I'll go back and manually find/replace instances of processor with prepper that the IDE missed.

File: data-prepper-api/src/test/java/com/amazon/dataprepper/model/PluginTypeTests.java [hunk 4]
```diff
@@ -21,7 +21,7 @@ public void testPluginTypeEnumValues() {
         checkPluginTypeValues(bufferPluginType, "buffer", Buffer.class);
 
         final PluginType processorPluginType = PluginType.PROCESSOR;
```
Comment (yadavcbala) [inline data-prepper-api/src/test/java/com/amazon/dataprepper/model/PluginTypeTests.java:23]: Assuming this will be in next iteration

**Pull request merged**